convert2rhel/utils.py

```diff
@@ -453,4 +454,26 @@ def backup(self):
             loggerinst.warning("Can't access %s" % TMP_DIR)
 
 
+def parse_version_number(version_str):
+    """
+    Parses the version number from the version string in /etc/system-release.
+    Returns a dict with major, minor, and release keys where each key is a number or None
+    """
+    if not isinstance(version_str, str):
+        raise TypeError("Expected version_str to be a string")
+
+    version_full_number = re.search(r"([\d.]+)", version_str).group()
+
+    version_number_regex = re.search(r"^(\d+)?\.?(\d+)?\.?(\*|\d+)?$", version_full_number)
+
+    if version_number_regex is None:
+        raise Exception("Could not find Operating System version number in \"" + version_str + "\"")
+
+    version_dict = {'major': version_number_regex.group(1) if version_number_regex.group(1) else None,
+                    'minor': version_number_regex.group(2) if version_number_regex.group(2) else None,
+                    'release': version_number_regex.group(3) if version_number_regex.group(3) else None}
```

Comment:
Maybe it'd make more sense if it returned as number and not string
```suggestion
    version_dict = {'major': int(version_number_regex.group(1)) if version_number_regex.group(1) else None,
                    'minor': int(version_number_regex.group(2)) if version_number_regex.group(2) else None,
                    'release': int(version_number_regex.group(3)) if version_number_regex.group(3) else None}
```